src/liblsquic/lsquic_full_conn_ietf.c

```diff
@@ -6186,6 +6186,21 @@ process_max_stream_data_frame (struct ietf_full_conn *conn,
         return 0;
     }
 
+    if ((conn->ifc_flags & (IFC_SERVER|IFC_HTTP)) == IFC_HTTP
```

Comment:
It is better to move those two tests under `if (is_peer_initiated(conn, stream_id))`, just in case side effects for regular MAX_STREAM_DATA processing.